### DISCUSSION TITLE
get points of intersection with the window’s boundary

### DISCUSSION THREAD

**Discussion by Aomandeyi** [CGAL/cgal#5914] [Q&A] [answered]
excuse me, could anyone give me some advice about this problem？

![image](https://user-images.githubusercontent.com/56006096/129829810-d72480e2-4567-441a-8163-24aebd02d221.png)

**Comment by sloriot** [comment, marked as answer]
You can use the function [`draw_dual()`](https://doc.cgal.org/latest/Triangulation_2/classCGAL_1_1Delaunay__triangulation__2.html#a7904b0176d781671b9805dfc599128fd) from the triangulation.

Then use the `intersection()` function to clip the segments, rays and lines.

It is used in the CGAL ipelets here:
https://github.com/CGAL/cgal/blob/master/CGAL_ipelets/include/CGAL/CGAL_Ipelet_base_v7.h#L724

which is calling this function:
https://github.com/CGAL/cgal/blob/master/CGAL_ipelets/include/CGAL/CGAL_Ipelet_base_v7.h#L705

Here is the struct used as stream for `draw_dual()`:
https://github.com/CGAL/cgal/blob/master/CGAL_ipelets/include/CGAL/CGAL_Ipelet_base_v7.h#L671
